src/interfaces/IBlocklockSender.sol

```diff
@@ -1,45 +1,71 @@
 // SPDX-License-Identifier: MIT
-pragma solidity 0.8.24;
-
-import "../libraries/TypesLib.sol";
-
-interface IBlocklockSender {
-    /**
-     * @notice Requests the generation of a blocklock decryption key at a specific blockHeight.
-     * @dev Initiates a blocklock decryption key request.
-     * The blocklock decryption key will be generated once the chain reaches the specified `blockHeight`.
-     * @return requestID The unique identifier assigned to this blocklock request.
-     */
-    function requestBlocklock(uint256 blockHeight, TypesLib.Ciphertext calldata ciphertext)
+pragma solidity ^0.8;
+
+import {TypesLib} from "../libraries/TypesLib.sol";
+
+import {ISubscription} from "../interfaces/ISubscription.sol";
+
+/// @title IBlocklockSender interface
+/// @author Randamu
+/// @notice Interface for periphery smart contract used to interact with the decryption sender contract.
+interface IBlocklockSender is ISubscription {
+    /// @notice Requests the generation of a blocklock decryption key at a specific blockHeight.
+    /// @dev Initiates a blocklock decryption key request.
+    /// The blocklock decryption key will be generated once the chain reaches the specified `blockHeight`.
+    /// @return requestID The unique identifier assigned to this blocklock request.
+    function requestBlocklockWithSubscription(
+        uint32 callbackGasLimit,
+        uint256 subId,
+        uint256 blockHeight,
+        TypesLib.Ciphertext calldata ciphertext
+    ) external payable returns (uint256 requestID);
+
+    /// @notice Requests a blocklock for a specified block height with the provided ciphertext without a subscription ID.
+    /// Requires payment to be made for the request without a subscription.
+    /// @param callbackGasLimit The gas limit allocated for the callback execution after the blocklock request
+    /// @param blockHeight The block height at which the blocklock is requested
+    /// @param ciphertext The ciphertext that will be used in the blocklock request
+    /// @return requestID The unique identifier for the blocklock request
+    /// @dev This function allows users to request a blocklock for a specific block height. The blocklock is not associated with any subscription ID
+    ///      and requires a ciphertext to be provided. The function checks that the contract is configured and not disabled before processing the request.
+    function requestBlocklockWithoutSubscription(
```

Comment:
I'd just name this `requestBlocklock` - people with a subscription will know about it, people without a subscription won't know what this means